Add license

kequach opened the pull request:
As discussed in #5, this PR adds a license to this repository. The license has been discussed with Erwin Leufkens, art director at UU.

jkorb commented:
Sorry it took me forever to get back to this for research reasons. I'm now hoping to get this merge finished soon. So:

First of all, looks great! Thanks so much for doing this. Three things:

+ There are small wording differences in the license and the MIT license I find on Wikipedia. Here's the diff:

   ```
    < use, copy, modify, merge, publish, and/or distribute, and to permit persons to
    < whom the Software is furnished to do so, subject to the following conditions:
    ---
    > use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
    > of the Software, and to permit persons to whom the Software is furnished to do
    > so, subject to the following conditions:
   ```

    Is this intentional? I'm not sure which license is authorative.

+ We should update the preambles of each class to reflect this (update the year and that we're now covered by MIT). The preamble of the new `uubeamer` class (:1st_place_medal:) can be an example. It'd be great if we could take care of this in this PR. Would you mind updating the files?

+ While you're at it, we can then delete the license file from the scripts folder, since it's automatically covered by this. 

+ Finally, we should probably add something about the copyrights of the pictures and logos. I.e. make clear that only the software is covered by the license. Perhaps a `README.md` in `lib/resources/`? Thoughts?

If you can't get around to doing these things, just let me know and I'll try next week. For this week, I'm all LaTeX'ed out after finishing the `uubeamer`` class :)

kequach commented:
Hi @jkorb, no worries about the response time. Yes, the wording differences are intentional, it's a modified MIT license. I'm not quite sure how to best describe this in the files, maybe something like MIT (modified)? I also asked  @J535D165 for some thoughts about this.
I updated and deleted the files according to the 2 bullet points. Regarding the pictures and logos, I think that is a good idea. However, I don't know anything about licensing for these. @J535D165 do you have a suggestion?

jkorb commented:
Awesome, thanks @kequach ! So, if I understand correctly, you guys took out the sublicensing and selling parts of MIT. Fair. But I guess, it's a mistake, then, to reference this as MIT, as I did in the `uubeamer` class. 

In principle, I'm cool with using a modified MIT as you suggest, I'm just a bit worried about confusions that may arise from using a non-standard license. Have you guys thought about that?

Might be a curveball, and perhaps best be ignored, but if we want to exclude commercial use (not that that'll be an issue), why don't we go with some standard non-commercial license, like the LaTeX license: https://www.latex-project.org/lppl/

I like MIT for its simplicity and because I personally don't care much about what people do with my work, I'm just happy if it helps. But that advantage, simplicity, might be lost by modyfying.

kequach commented:
I agree with the issues in using a modified license. It also makes it difficult to immediately understand what was modified. In that case I'd wait for a response from @J535D165, as he was the one who aligned this with Erwin Leufkens.

jkorb commented:
Fair! Let's see what he says :)

jkorb commented:
Great that you're chiming in, @J535D165 ! And thanks for hashing this out with Erwin.

I'm cool with publishing my templates under CC0. As I said, I'm not really keen on recognition and like to keep things simple. So public domain sounds like a good idea :)  And if I understand correctly, this should be cool with Erwin/the Corporate Identity department, since it just covers the code and not the branding it implements, right? 

So, how would we realize this: CC0 license disclaimer everywhere together with the existing copyright disclaimers about the branding? I would also put my name, email, and a link to this repo in there, not for copyright reasons but so that people can find the most recent version. 

Do we need to double check this with Erwin? 

Once the copyright issue is solved, I'll be looking into packaging this into an official release for ease of use.

jkorb commented:
Hi @J535D165 and @tymees ! I'm really sorry that this fizzled, but I'm keen on picking this back up. Are you still around? So, my favorite option would be CC0 as mentioned. The only open questioin would be whether this needs to be checked with Erwin.

tymees commented:
Not entirely sure why you've mentioned me, my only contribution here is so small I doubt it would even be eligible for copyright ;)

Anyway, CC0 looks fine for my one-word-change ;)

Just to weigh in on the branding stuff;
First of, I am not a lawyer, I'm just parroting someone I consulted when my team ran into a pretty much the same situation in a different UU-branding related project.

A lot of the branding is (also) protected by trademark law, which the license does not cover. So, there is additional protection there for the branding. (In addition to other copyright related stuff that might still be applicable, as the license should indeed cover the code itself only).

J535D165 commented:
Agreed. There is no need to check this with Erwin. The contact I had with him should be fine. 

Update this PR to CC0?

jkorb commented:
Sorry, @tymees ! Tagging suggestions confused me, but that's very valuable input, so thanks for that! 

I meant to tag @kequach , since this is his PR/initiative.

@J535D165 : awesome, thanks! Yes, let's move to CC0. 

@kequach can you update the PR (if you're still interested)? Otherwise, I can take care of it this week.

kequach commented:
@jkorb please have a look if the changes are fine and feel free to make any suggestions

jkorb commented:
Awesome, thanks so much!! I'll go through this tomorrow!

kequach commented:
I didn't do that much but you're welcome 😀

jkorb commented:
Hi @kequach @J535D165 (and @tymees ). I've updated the PR to follow the recommendations on using CC0 for software by the CC-peeps (https://wiki.creativecommons.org/wiki/CC0_FAQ#May_I_apply_CC0_to_computer_software.3F_If_so.2C_is_there_a_recommended_implementation.3F). I'd be cool with merging this now. Does anybody want to have a second look? Would be awesome for my peace of mind :)

jkorb commented:
Thanks, @kequach !

kequach commented on README.md:
```diff
@@ -60,8 +60,26 @@ you can use `uucls` in the command line to manage the package.
+
+This software was originally written by Johannes Korbmacher
+([j.korbmacher@uu.nl](mailto:j.korbmacher@uu.nl)) and contains contributions
+from Ty Mees ([t.d.mees@uu.nl](mailto:t.d.mees@uu.nl)) and Kevin Quach.
```
```suggestion
from Ty Mees ([t.d.mees@uu.nl](mailto:t.d.mees@uu.nl)) and Keven Quach.
```

jkorb replied:
Uff, sorry for mispelling your name. I wasn't paying due dilligence!

kequach commented on COPYING:
I would suggest to keep the file name as "LICENSE", since this allows GitHub to automatically detect the license. COPYING is not a machine-readable file name on GitHub, even though CC0 recommends this naming.

jkorb replied:
Ah good to know. Let's go with that.

kequach commented:
I added two remarks. Besides that, I think we are ready to merge @jkorb

jkorb commented:
> I added two remarks. Besides that, I think we are ready to merge @jkorb

Awesome! Thanks so much, @kequach for starting this and @J535D165 , @tymees for chiming in. 

If you're OK with that, @kequach , I'll reorder some commits and ```push --force``` to this PR/your fork's main branch. This would be a horrible idea if you have local work on the repo on your computer, but I assume you don't, right? If you don't have local work, it would be great if I could re-order and better document some of the commits in this way :)

kequach commented:
> > I added two remarks. Besides that, I think we are ready to merge @jkorb
> 
> Awesome! Thanks so much, @kequach for starting this and @J535D165 , @tymees for chiming in. 
> 
> If you're OK with that, @kequach , I'll reorder some commits and ```push --force``` to this PR/your fork's main branch. This would be a horrible idea if you have local work on the repo on your computer, but I assume you don't, right? If you don't have local work, it would be great if I could re-order and better document some of the commits in this way :)

Feel free to go ahead!
